styleguide/writing-style.md

```diff
@@ -0,0 +1,183 @@
+# Writing style guide
+
+Welcome to the Buildkite writing style guide.
+
+These guidelines provide details about the language and words used to write the Buildkite docs, as well as details about its writing style and format.
+If something isn't included in this guide, see the [Google developer documentation style guide](https://developers.google.com/style), followed by the [Microsoft Style Guide](https://docs.microsoft.com/en-us/style-guide/welcome/).
+
+For details about the Markdown syntax used to render your writing as well as the Buildkite docs' file structure, as well as details on working with the site, and using screenshots, refer to the [Markdown syntax style guide](markdown-syntax-style.md).
+
+Table of contents (main headings):
+* [Language](#language)
```

Comment:
Nitpicks:

- Add a blank line before the start of a list.
- "(main headings)" feels unnecessary to me.
- I have a preference for `-` in lists, but I can change. 😅

Reply:
> Add a blank line before the start of a list.

Whoops, well spotted!

> "(main headings)" feels unnecessary to me.

Yeah, my main reasoning here was to make the distinction between this list and GitHub's auto-generated outline. I've removed it.

> I have a preference for `-` in lists, but I can change. 😅

Sorry, I thought you were referring to the lead-in to a list rather than the Markdown syntax symbol for a bullet point list item, which I think is what you meant here lol

For the lead in, it looks like the [Google style guide](https://developers.google.com/style) tend to implement colons for these, although I think we're on the same page with this one.

For the Markdown syntax symbol, I've noticed more instances of an asterisk `*` than a hyphen/minus `-`, so maybe we should choose asterisks `*`? That way, it'll be easier to update the docs at a later point in time.

Note to self—I should add this to the Markdown syntax style guide about bulleted lists.

Reply:
Michael 😢 and I agreed to stick to asterisks due to reason above.

Reply:
This is all now covered.

Reply:
Darn, I've only just discovered from the linting results that the first bullet point must be a hyphen `-` and nested can then be asterisks `*` 😲 

I'll update the guidance accordingly 🤦